hardware/DomoticzTCP.h

```diff
@@ -47,10 +45,8 @@ class DomoticzTCP : public CRFXBase
 	std::string m_password;
 	int m_retrycntr;
 	std::shared_ptr<std::thread> m_thread;
-	sockaddr_in6 m_addr;
-	struct addrinfo *info;
-	int m_socket;
-	unsigned char mBuffer[512];
+	csocket *m_connection;
```

Comment:
I'm not a big fan of new/delete usages.... It saves a header inclusion ...
Another approach would be to inherit csocket

class DomoticzTCP : public CRFXBase, csocket

I am not a bit fan of the class name 'csocket' and member functions at the moment... 
Not even sure it is working for IPV6 (as this class (DomoticzTCP) was working with

I am a fan of ASyncTCP, this is a much better class. Is it possible to use this instead ?